src/partition/test-repart.sh

```diff
@@ -222,3 +222,49 @@ echo "### Testing json output ###"
 "$repart" "$D/zzz" --size=3G --dry-run=no --seed="$SEED" --definitions="$D/definitions" --no-pager --json=help
 "$repart" "$D/zzz" --size=3G --dry-run=no --seed="$SEED" --definitions="$D/definitions" --no-pager --json=pretty
 "$repart" "$D/zzz" --size=3G --dry-run=no --seed="$SEED" --definitions="$D/definitions" --no-pager --json=short
+
+echo "### Testing drop-in overrides ###"
+
+mkdir -p "$D/definitions-overrides"
+
+cat >"$D/definitions-overrides/root.conf" <<EOF
+[Partition]
+Type=swap
+SizeMaxBytes=64M
+UUID=837c3d67-21b3-478e-be82-7e7f83bf96d3
+EOF
+
+mkdir -p "$D/definitions-overrides/root.conf.d"
+
+cat >"$D/definitions-overrides/root.conf.d/override1.conf" <<EOF
+[Partition]
+Label=label1
+SizeMaxBytes=32M
+EOF
+
+cat >"$D/definitions-overrides/root.conf.d/override2.conf" <<EOF
+[Partition]
+Label=label2
+EOF
+
+rm -f abc
+
+systemd-repart --definitions="$D/definitions-overrides" --dry-run=yes --empty=create --size=100M --json=pretty abc >"$D/dropin-overrides"
+
+cmp "$D/dropin-overrides" - <<EOF
```

Comment:
You can use `diff` instead of `cmp`, it provides better output when things don't match.

To pass strings to `diff`, you can do the following:

```
diff <(echo "$JSON_OUTPUT") - <<EOF
<json>
EOF
```